### ISSUE TITLE
ng lint with --format=json generates invalid JSON when no linting errors were found

### ISSUE THREAD

**Issue by dsereno** [angular-eslint/angular-eslint#383]
After the latest update the lint report given by the command ng lint --format=json > report.json generates invalid JSON file when there are no lint errors reported in the code

🐞 Bug report
Description
When there are no errors found in the the ng lint --force=true --silent=true --format=json > report.json command the file created is empty which is parsed as an invalid JSON.

Expected behaviour:
Result: 
    []

Actual Result:

Additionally not using silent=true will also result in an invalid JSON
Expected behaviour:
Result: 
    []

Actual Result:
    Linting "test-app"...
    All files pass linting.

🔬 Minimal Reproduction
ng new eslint-report - create a new angular project
Add some correct code
ng lint --force=true --silent=true --format=json > report.json
Open report.json to see invalid JSON
🔥 Exception or Error
report.json
    ""
🌍 Your Environment

     _                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 11.2.3
Node: 13.12.0
OS: darwin x64

Angular: 11.2.4
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1102.3
@angular-devkit/build-angular   0.1102.3
@angular-devkit/core            11.2.3
@angular-devkit/schematics      11.2.3
@angular/cli                    11.2.3
@schematics/angular             11.2.3
@schematics/update              0.1102.3
rxjs                            6.6.6
typescript                      4.1.5

**Comment by JamesHenry** [comment]
@orangedog I'm guessing for checkstyle would you also expect to always receive the xml output? Rather than the empty output you currently get?

E.g. would you rather get this:

```
→ npx ng lint allergies-fusion --format=checkstyle --silent
<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"></checkstyle>
```

If that's the case then we can switch the builder to always call format() on the results. For the common case it will be a blank line, for custom formatters it will be based on whatever that formatter does with no results.

**Comment by JamesHenry** [comment]
Out of an abundance of caution I'm going to class this as a breaking change, because I want to do a v3 release anyway. Please let me know your thoughts @orangedog I am not a user of custom formatters myself so the more info the better

**Comment by OrangeDog** [comment]
An empty output does not cause me any issues, but a valid xml reporting no issues would probably be better.
